pybamm/models/full_battery_models/lithium_ion/electrode_soh.py

```diff
@@ -557,3 +557,54 @@ def get_min_max_stoichiometries(
     """
     esoh_solver = ElectrodeSOHSolver(parameter_values, param, known_value)
     return esoh_solver.get_min_max_stoichiometries()
+
+
+def calculate_theoretical_energy(
+    parameter_values, initial_soc=1.0, final_soc=0.0, points=100
+):
+    """
+    Calculate maximum energy possible from a cell given OCV, initial soc, and final soc
+    given voltage limits, open-circuit potentials, etc defined by parameter_values
+
+    Parameters
+    ----------
+    parameter_values : :class:`pybamm.ParameterValues`
+        The parameter values class that will be used for the simulation.
+    initial_soc : float
+        The soc at begining of discharge, default 1.0
+    final_soc : float
+        The soc at end of discharge, default 1.0
+    points : int
+        The number of points at which to calculate voltage.
+
+    Returns
+    -------
+    E
+        The total energy of the cell in Wh
+    """
+    # Get initial and final stoichiometric values.
+    n_i, p_i = get_initial_stoichiometries(initial_soc, parameter_values)
+    n_f, p_f = get_initial_stoichiometries(final_soc, parameter_values)
+    n_vals = np.linspace(n_i, n_f, num=points)
+    p_vals = np.linspace(p_i, p_f, num=points)
+    # Calculate OCV at each stoichiometry
+    Vs = np.empty(n_vals.shape)
+    for i in range(n_vals.size):
+        V_pos = parameter_values["Positive electrode OCP [V]"](p_vals[i]).value
+        V_neg = parameter_values["Negative electrode OCP [V]"](n_vals[i]).value
+        Vs[i] = V_pos - V_neg
```

Comment:
Something like this should work and be cleaner (it will also work with different temperatures and interpolant OCVs this way)
```python
T = param.T_amb
n_vals = pybamm.linspace(n_i, n_f, num=points)`
p_vals = pybamm.linspace(p_i, p_f, num=points)`
Vs = parameter_values.evaluate(param.p.prim.U(p_vals, T)- param.n.prim.U(n_vals, T))
```